internal/app/osd/internal/reg/reg.go

```diff
@@ -217,15 +218,16 @@ func (r *Registrator) Reset(ctx context.Context, in *empty.Empty) (reply *proto.
 		),
 	)
 
-	err = cr.Open()
+	err = cr.Open(context.Background())
 	if err != nil {
 		return nil, err
 	}
 
 	// nolint: errcheck
 	defer cr.Close()
 
-	err = cr.Run()
+	// TODO: should this go through system.Services?
```

Comment:
That is exactly what I want to do. I would like `reset` to be a service. Or we allow overriding the arguments to the `kubeadm` service. We can address in a followup PR.

Reply:
yep, I think we can inject that as yet another service, I just need to fix `system.Services()` to allow to start finished/failed services once again, but that's easy.